security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java

```diff
@@ -107,8 +107,7 @@ public Object loadUserBySAML(SAMLCredential credential)
             if (attrName.equals(SAML_IDP_METADATA_ROLE_ATTR_NAME)) {
                 List<XMLObject> attributeValues = cAttribute.getAttributeValues();
                 if (!attributeValues.isEmpty()) {
-                    userRoles.add(new StringBuilder(APP_NAME).append(":").append(
-                        getAttributeValue(attributeValues.get(0))).toString());
+                    userRoles.add(getAttributeValue(attributeValues.get(0)));
```

Comment:
That should work in theory. Keycloak cloud server may be better solution for your scenario, in which you can have a dedicated keycloak for each client.

At PM, the purpose of choosing keylcoak is to create a user federation to unify the users from different institutes to have one portal which shows a different view for each individual base on the roles granted. Our clinicians had hard time to remember which patients in which portals for which version of datasets.